File: Simple Linear Regression/simple_linear_regression.py
#Data Preprocessing

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

from sklearn.cross_validation import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=1/3,random_state=0)

#Fitting Simple Linear Regression to Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor = regressor.fit(X_train,y_train)

#Predicting the Test Set results
y_pred = regressor.predict(X_test)

#Visualising the training set results
plt.scatter(X_train,y_train, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title("Salary vs Years of Experience(Training set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

#Visualising the test set results
plt.scatter(X_test,y_test, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title("Salary vs Years of Experience(Test set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

File: Regression models/Simple Linear Regression/simple_linear_regression.R
#Data Preprocessing

dataset = read.csv('Salary_Data.csv')

library(caTools)
set.seed(123)

split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)

#Fitting Simple Linear Regression to Training set
regressor = lm(formula = Salary ~ YearsExperience, training_set)

#Predicting the Test Set results
y_pred = predict(regressor  , newdata = test_set)

#Visualising the Training set result
#install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes( x = training_set$YearsExperience, y = training_set$Salary),
                  color = 'red') +
  geom_line(aes( x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Years of Experience(Training set)') +
  xlab("Years of Experience") +
  ylab("Salary")

ggplot() +
  geom_point(aes( x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') +
  geom_line(aes( x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Years of Experience(Test set)') +
  xlab("Years of Experience") +
  ylab("Salary")



             